src/metabase/lib/join.cljc

```diff
@@ -455,9 +470,39 @@
     _lhs-column-or-nil :- [:maybe lib.metadata/ColumnMetadata]
     _rhs-column-or-nil :- [:maybe lib.metadata/ColumnMetadata]]
    ;; currently hardcoded to these six operators regardless of LHS and RHS.
-   [{:lib/type :mbql.filter/operator, :short :=, :display-name  (i18n/tru "Equal to")}
-    {:lib/type :mbql.filter/operator, :short :>, :display-name  (i18n/tru "Greater than")}
-    {:lib/type :mbql.filter/operator, :short :<, :display-name  (i18n/tru "Less than")}
-    {:lib/type :mbql.filter/operator, :short :>=, :display-name (i18n/tru "Greater than or equal to")}
-    {:lib/type :mbql.filter/operator, :short :<=, :display-name (i18n/tru "Less than or equal to")}
-    {:lib/type :mbql.filter/operator, :short :!=, :display-name (i18n/tru "Not equal to")}]))
+   (join-condition-operator-definitions)))
+
+(mu/defn ^:private pk-column :- [:maybe lib.metadata/ColumnMetadata]
+  "Given something `x` (e.g. a Table metadata) find the PK column."
+  [query        :- ::lib.schema/query
+   stage-number :- :int
+   x]
+  (m/find-first (fn [{:keys [semantic-type], :as _col}]
+                  (isa? semantic-type :type/PK))
+                (lib.metadata.calculation/visible-columns query stage-number x)))
+
+(mu/defn ^:private fk-column :- [:maybe lib.metadata/ColumnMetadata]
+  "Given a query stage find an FK column that points to the PK `pk-col`."
+  [query        :- ::lib.schema/query
+   stage-number :- :int
+   pk-col       :- [:maybe lib.metadata/ColumnMetadata]]
+  (when-let [pk-id (:id pk-col)]
+    (m/find-first (fn [{:keys [semantic-type fk-target-field-id], :as _col}]
+                    (and (isa? semantic-type :type/FK)
```

Comment:
I wonder if we should prefer the functions like `(metabase.lib.types.isa/foreign-key? col)` over direct checks like this. They are overridable should the need arise.